src/sycl/sycl_compat.hpp

```diff
@@ -40,6 +40,14 @@ template <typename native_object_t, typename sycl_object_t>
 native_object_t get_native(const sycl_object_t &sycl_object) {
     return reinterpret_cast<native_object_t>(get_native(sycl_object));
 }
+#ifdef DNNL_SYCL_CUDA
+template <>
+CUcontext get_native(const ::sycl::device &device);
+#endif
+#ifdef DNNL_SYCL_HIP
+template <>
+HIPcontext get_native(const ::sycl::device &device);
+#endif
```

Comment:
This file should not contain CUDA/HIP specific code, can you please move it to `sycl_cuda_compat.hpp` and `sycl_hip_compat.hpp`?

Reply:
Will do

Reply:
Change made